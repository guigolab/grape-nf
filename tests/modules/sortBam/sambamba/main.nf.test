nextflow_process {

    name "Test Process sortBam"
    script "modules/sortBam/sambamba/main.nf"
    process "sortBam"

    tag "module"

    test("Should sort BAM file with sambamba") {

        when {
            process {
                """
                input[0] = Channel.from([
                    [ "sample3", "test3", file("${baseDir}/data/sample3_m4_n10_toGenome.bam"), "bam", "GenomeAlignments", true]
                ])
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
