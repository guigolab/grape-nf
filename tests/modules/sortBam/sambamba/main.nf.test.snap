{
    "Should sort BAM file with sambamba": {
        "content": [
            {
                "0": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome_sorted.bam:md5,9b72491669335addf88cf1a631b6e5d1",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ],
                "1": [
                    [
                        "sample3",
                        "test3",
                        "sample3_m4_n10_toGenome_sorted.bam.bai:md5,2a87071e7749c3c170e52def44795307",
                        "bam",
                        "GenomeAlignments",
                        true
                    ]
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-25T16:39:05.913863"
    }
}